_episodes/debug.md

```diff
@@ -42,75 +42,16 @@ Some very common YAML errors are:
 
 - Using tabs instead of spaces. In YAML files indentations are made using spaces, not tabs.
   Errors caused by tabs will show `'NoneType' object has no attribute 'name'`.
+  Please download and run [this example][tab-error] which includes a tab character.
 
   ~~~
-  cwlVersion: v1.2
-  class: Workflow
-
-  inputs:
-    rna_reads_human: File
-   ref_genome: Directory
-   annotations: File
-
-  steps:
-   quality_control:
-   run: bio-cwl-tools/fastqc/fastqc_2.cwl
-   in:
-   reads_file: rna_reads_human
-   out: [html_file]
-
-  mapping_reads:
-   requirements:
-   ResourceRequirement:
-   ramMin: 9000
-   run: bio-cwl-tools/STAR/STAR-Align.cwl
-   in:
-   RunThreadN: {default: 4}
-   GenomeDir: ref_genome
-   ForwardReads: rna_reads_human
-   OutSAMtype: {default: BAM}
-   SortedByCoordinate: {default: true}
-   OutSAMunmapped: {default: Within}
-   out: [alignment]
-
-  index_alignment:
-   run: bio-cwl-tools/samtools/samtools_index.cwl
-   in:
-   bam_sorted: mapping_reads/alignment
-   out: [bam_sorted_indexed]
-
-  count_reads:
-   requirements:
-   ResourceRequirement:
-   ramMin: 500
-   run: bio-cwl-tools/subread/featureCounts.cwl
-   in:
-   mapped_reads: index_alignment/bam_sorted_indexed
-   annotations: annotations
-   out: [featurecounts]
-
-  outputs:
-  qc_html:
-   type: File
-   outputSource: quality_control/html_file
-   bam_sorted_indexed:
-   type: File
-   outputSource: index_alignment/bam_sorted_indexed
-   featurecounts:
-   type: File
-   outputSource: count_reads/featurecounts
-  ~~~
-  {: .language-yaml}
-
-  ~~~
-  $ cwltool rna_seq_workflow.cwl workflow_input.yml
+  $ cwltool tab-error.cwl workflow_input.yml
   ~~~
   {: .language-bash}
 
   ~~~
-ERROR Tool definition failed validation:
-while scanning for the next token
-tabs_rna_seq_workflow.cwl:6:1:   found character '\t' that cannot start any token
+  ERROR I'm sorry, I couldn't load this CWL file, try again with --debug for more information.
+  The error was: 'NoneType' object has no attribute 'name
```

Comment:
What is your `cwltool` and `schema-salad` version? I released updates that match the error message I committed in https://github.com/carpentries-incubator/cwl-novice-tutorial/commit/8f3f5703bd780014253fbb04ebe241c0832937cc

Reply:
Maybe we should update the installation instructions to ensure that learners have a minimum version of `schema-salad` so that they get the same experience

Reply:
The fix is in https://github.com/common-workflow-language/schema_salad/commit/04a3c7240ae8352bfc21fdcafada3ee433c7827a part of the https://github.com/common-workflow-language/schema_salad/releases/tag/8.2.20220204150214 release

Reply:
$ cwltool --version
/home/mbexegc2/Documents/projects/bioexcel/follow-cwl-novice-tutorial/venv/bin/cwltool 3.1.20211107152837
$ schema-salad-tool --version
/home/mbexegc2/Documents/projects/bioexcel/follow-cwl-novice-tutorial/venv/bin/schema-salad-tool Current version: 8.2.20220103095339

I'll recreate my venv and see where that gets me.

Reply:
Yep, that's got me the same error.

Reply:
I just made a new virtualenv and `pip install cwltool` which got me

```
$ pip show cwltool | grep Version
Version: 3.1.20220204090313
$ pip show schema-salad | grep Version
Version: 8.2.20220204150214
$ cwltool code/tab-error.cwl
INFO /home/michael/cwl-novice-tutorial/env/bin/cwltool 3.1.20220204090313
INFO Resolved 'code/tab-error.cwl' to 'file:///home/michael/cwl-novice-tutorial/code/tab-error.cwl'
ERROR Tool definition failed validation:
while scanning for the next token
code/tab-error.cwl:5:1:   found character '\t' that cannot start any token
```
So I'm a bit confused as to why you see something different.
<details><summary>`pip freeze`</summary>
```
argcomplete==2.0.0
bagit==1.8.1
CacheControl==0.12.10
certifi==2021.10.8
charset-normalizer==2.0.11
coloredlogs==15.0.1
cwltool==3.1.20220204090313
humanfriendly==10.0
idna==3.3
isodate==0.6.1
lockfile==0.12.2
lxml==4.7.1
mistune==0.8.4
msgpack==1.0.3
mypy-extensions==0.4.3
networkx==2.6.3
prov==1.5.1
psutil==5.9.0
pydot==1.4.2
pyparsing==3.0.7
python-dateutil==2.8.2
rdflib==6.1.1
requests==2.27.1
ruamel.yaml==0.17.20
ruamel.yaml.clib==0.2.6
schema-salad==8.2.20220204150214
shellescape==3.8.1
six==1.16.0
typing_extensions==4.0.1
urllib3==1.26.8
```
</details>

Reply:
rebase fail (user error). I'll fix it and update this branch.